packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```diff
@@ -299,9 +324,376 @@ describe('EscrowClient', () => {
+        recordingOracle: ethers.ZeroAddress,
+        reputationOracle: ethers.ZeroAddress,
+        exchangeOracle: ethers.ZeroAddress,
+        recordingOracleFee: 10n,
+        reputationOracleFee: 10n,
+        exchangeOracleFee: 0n,
+        manifest: VALID_URL,
+        manifestHash: FAKE_HASH,
+      };
+
+      await expect(
+        escrowClient.createFundAndSetupEscrow(
+          tokenAddress,
+          10n,
+          jobRequesterId,
+          escrowConfig
+        )
+      ).rejects.toThrow(ErrorAmountMustBeGreaterThanZero);
+    });
+
+    test('should throw an error if total fee > 100', async () => {
+      const escrowConfig = {
+        recordingOracle: ethers.ZeroAddress,
+        reputationOracle: ethers.ZeroAddress,
+        exchangeOracle: ethers.ZeroAddress,
+        recordingOracleFee: 40n,
+        reputationOracleFee: 40n,
+        exchangeOracleFee: 40n,
+        manifest: VALID_URL,
+        manifestHash: FAKE_HASH,
+      };
+
+      await expect(
+        escrowClient.createFundAndSetupEscrow(
+          tokenAddress,
+          10n,
+          jobRequesterId,
+          escrowConfig
+        )
+      ).rejects.toThrow(ErrorTotalFeeMustBeLessThanHundred);
+    });
+
+    test('should throw an error if manifest is an empty string', async () => {
+      const escrowConfig = {
+        recordingOracle: ethers.ZeroAddress,
+        reputationOracle: ethers.ZeroAddress,
+        exchangeOracle: ethers.ZeroAddress,
+        recordingOracleFee: 10n,
+        reputationOracleFee: 10n,
+        exchangeOracleFee: 10n,
+        manifest: '',
+        manifestHash: FAKE_HASH,
+      };
+
+      await expect(
+        escrowClient.createFundAndSetupEscrow(
+          tokenAddress,
+          10n,
+          jobRequesterId,
+          escrowConfig
+        )
+      ).rejects.toThrow(ErrorInvalidManifest);
+    });
+
+    test('should throw an error if hash is an empty string', async () => {
+      const escrowConfig = {
+        recordingOracle: ethers.ZeroAddress,
+        reputationOracle: ethers.ZeroAddress,
+        exchangeOracle: ethers.ZeroAddress,
+        recordingOracleFee: 10n,
+        reputationOracleFee: 10n,
+        exchangeOracleFee: 10n,
+        manifest: VALID_URL,
+        manifestHash: '',
+      };
+
+      await expect(
+        escrowClient.createFundAndSetupEscrow(
+          tokenAddress,
+          10n,
+          jobRequesterId,
+          escrowConfig
+        )
+      ).rejects.toThrow(ErrorHashIsEmptyString);
+    });
+
+    test('should throw an error if the createFundAndSetupEscrow fails', async () => {
+      const jobRequesterId = 'job-requester';
+
+      escrowClient.escrowFactoryContract.createFundAndSetupEscrow.mockRejectedValueOnce(
+        new Error()
+      );
+
+      await expect(
+        escrowClient.createFundAndSetupEscrow(
+          tokenAddress,
+          10n,
+          jobRequesterId,
+          escrowConfig
+        )
+      ).rejects.toThrow();
```

Comment:
nit (here and for `createEscrow`
Would be also nice to check that it fails with correct error (since under the hood it's wrapped into `throwError` call

Reply:
https://github.com/humanprotocol/human-protocol/pull/3641/commits/3867c517fa914747440b18d5a0f2bff7a7c7171c